onnxruntime/core/providers/webnn/builders/impl/rotaryEmbedding_op_builder.cc

```diff
@@ -109,9 +109,18 @@ Status RotaryEmbeddingOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_build
     rotary_embedding_dim = head_size;
   }
 
-  // First ensure the input has shape (batch_size, num_heads, sequence_length, head_size).
-  if (!input_is_4d) {
-    const std::vector<uint32_t> new_shape{batch_size, num_heads, sequence_length, head_size};
+  emscripten::val transpose_options = emscripten::val::object();
+
+  // First ensure the input has shape (batch_size, sequence_length, num_heads, head_size).
```

Comment:
```suggestion
  // The input can be:
  // - 3D: [batch_size, sequence_length, hidden_size]
  // - 4D: [batch_size, num_heads, sequence_length, head_size]
  // Ensure the input is reshaped to: [batch_size, sequence_length, num_heads, head_size].
```

From the doc https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#com.microsoft.RotaryEmbedding.

Reply:
Thanks, done.